Makefile

```diff
@@ -63,6 +69,71 @@ check_headers:
 	@echo "Checking file headers"
 	@python test/verify_boilerplate.py
 
+# Integration tests
+.PHONY: test_integration
+test_integration:
+	bundle install
+	bundle exec kitchen create
+	bundle exec kitchen converge
+	bundle exec kitchen converge
+	bundle exec kitchen verify
+	bundle exec kitchen destroy
+
 .PHONY: generate_docs
 generate_docs:
 	@source test/make.sh && generate_docs
+
+# Versioning
+.PHONY: version
+version:
+	@source helpers/version-repo.sh
+
+# Run docker
+.PHONY: docker_run
+docker_run:
+	docker run --rm -it \
+		-e CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${CREDENTIALS_PATH} \
+		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
+		-v $(CURDIR):/cft/workdir \
+		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
+		/bin/bash
+
+.PHONY: docker_create
+docker_create:
+	docker run --rm -it \
+		-e CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${CREDENTIALS_PATH} \
+		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
+		-v $(CURDIR):/cft/workdir \
+		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
+		/bin/bash -c "kitchen create"
```

Comment:
Why is `bundle exec` necessary to add?

Reply:
This change doesn't work with the current docker image:

```
$ PAGER=cat git diff
diff --git i/Makefile w/Makefile
index c90f8ec..784e26e 100644
--- i/Makefile
+++ w/Makefile
@@ -105,7 +105,7 @@ docker_create:
                -e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
                -v $(CURDIR):/cft/workdir \
                ${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-               /bin/bash -c "kitchen create"
+               /bin/bash -c "bundle exec kitchen create"

 .PHONY: docker_converge
 docker_converge:
@@ -114,7 +114,7 @@ docker_converge:
                -e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
                -v $(CURDIR):/cft/workdir \
                ${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-               /bin/bash -c "kitchen converge && kitchen converge"
+               /bin/bash -c "bundle exec kitchen converge && bundle exec kitchen converge"

 .PHONY: docker_verify
 docker_verify:
@@ -123,7 +123,7 @@ docker_verify:
                -e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
                -v $(CURDIR):/cft/workdir \
                ${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-               /bin/bash -c "kitchen verify"
+               /bin/bash -c "bundle exec kitchen verify"

 .PHONY: docker_destroy
 docker_destroy:
@@ -132,7 +132,7 @@ docker_destroy:
                -e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
                -v $(CURDIR):/cft/workdir \
                ${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-               /bin/bash -c "kitchen destroy"
+               /bin/bash -c "bundle exec kitchen destroy"

 .PHONY: test_integration_docker
 test_integration_docker: docker_create docker_converge docker_verify docker_destroy
$ make docker_create docker_converge docker_verify docker_destroy
docker run --rm -it \
                -e CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=/cft/workdir/credentials.json \
                -e GOOGLE_APPLICATION_CREDENTIALS=/cft/workdir/credentials.json \
                -v /Users/thebo/lagrange/sources/github.com/terraform-google-modules/terraform-google-forseti:/cft/workdir \
                gcr.io/cloud-foundation-cicd/cft/kitchen-terraform:0.11.10_216.0.0_1.19.1_0.1.10 \
                /bin/bash -c "bundle exec kitchen create"
Could not find aws-sdk-core-2.11.205 in any of the sources
Run `bundle install` to install missing gems.
make: *** [docker_create] Error 7
```

For context, the Makefile is copied from the [module template Makefile](https://github.com/terraform-google-modules/terraform-google-module-template/blob/master/terraform-google-%7B%7Bcookiecutter.module_name%7D%7D/Makefile#L104-L138) and is configured accordingly - if we make changes here I think that they should land in the module template first and then propagate to other projects.

Reply:
@jeffmccune without `bundle exec`, there is no guarantee that the gems in the `Gemfile.lock` are invoked when `kitchen` is executed.

@adrienthebo The source of that problem is that the Dockerfile is installing the bundle under `/opt/kitchen`, rather than the `WORKDIR` of the container. I believe that change has already been requested for the template but it does not need to be a blocker here.